### PR TITLE
[Dependency Update] Bump up the CI Nvidia docker to CUDA 10.1

### PR DESCRIPTION
**Pull request by stu1130**
## Description ##
don't merge the code after we update the GPU driver of instances running the Jenkin
Bump up the Nvidia docker to CUDA 10.1

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

## Comments ##
@perdasilva @marcoabreu

### CODE REVIEW

File: ci/docker/Dockerfile.build.ubuntu_gpu_cu100 [hunk 1]
```diff
@@ -67,10 +67,6 @@ RUN /work/ubuntu_docs.sh
 COPY install/ubuntu_tutorials.sh /work/
 RUN /work/ubuntu_tutorials.sh
 
-ENV CUDNN_VERSION=7.3.1.20
-COPY install/ubuntu_cudnn.sh /work/
-RUN /work/ubuntu_cudnn.sh
```
Comment (marcoabreu) [inline ci/docker/Dockerfile.build.ubuntu_gpu_cu100]: ?

File: ci/docker/Dockerfile.build.ubuntu_gpu_cu101 [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+# Always last
+ARG USER_ID=0
+ARG GROUP_ID=0
+COPY install/ubuntu_adduser.sh /work/
+RUN /work/ubuntu_adduser.sh
+
+ENV CUDNN_VERSION=7.5.1.10
+COPY install/ubuntu_cudnn.sh /work/
+RUN /work/ubuntu_cudnn.sh
```
Comment (marcoabreu) [inline ci/docker/Dockerfile.build.ubuntu_gpu_cu101]: Please place this part before add user

File: ci/docker/Dockerfile.build.ubuntu_gpu_cu90 [hunk 3]
```diff
@@ -67,7 +67,6 @@ RUN /work/ubuntu_docs.sh
 COPY install/ubuntu_tutorials.sh /work/
 RUN /work/ubuntu_tutorials.sh
 
-ENV CUDNN_VERSION=7.3.1.20
```
Comment (marcoabreu) [inline ci/docker/Dockerfile.build.ubuntu_gpu_cu90]: ?

File: ci/docker/Dockerfile.build.ubuntu_gpu_cu100 (hunk 1 above)
Comment (stu1130) [reply ci/docker/Dockerfile.build.ubuntu_gpu_cu100]: On my previously PR, I resolved the commit by keeping two cudnn installation accidentally... let seperate this part and create another one to fix it

Comment (stu1130) [reply ci/docker/Dockerfile.build.ubuntu_gpu_cu100]: I created another PR #14996 to clean up the duplicate cudnn installation

**Comment by perdasilva** [comment]
Please don't for get to update the NightlyTestsForBinaries, for instance [here](https://github.com/apache/incubator-mxnet/blob/master/tests/nightly/JenkinsfileForBinaries#L38)

**Comment by stu1130** [comment]
Good catch @perdasilva done!

**Comment by pinaraws** [comment]
@mxnet-label-bot add[CI, pr-awaiting-response]

**Comment by marcoabreu** [comment]
@perdasilva waiting for your approval here

**Comment by perdasilva** [comment]
Damn. You probably also need to update: https://github.com/apache/incubator-mxnet/blob/master/ci/docker/install/ubuntu_nvidia.sh#L25

**Comment by perdasilva** [comment]
@stu1130 sorry this is taking longer than it should. The problem with the sanity check has been fixed. Could you please rebase off master to get everything to green =)

**Comment by stu1130** [comment]
@perdasilva done

**Comment by stu1130** [comment]
@marcoabreu @perdasilva I found a typo on setup_gpu_build_tools.sh please wait for that fix

File: tools/setup_gpu_build_tools.sh [hunk 4]
```diff
@@ -29,7 +29,7 @@ DEPS_PATH=$2
 >&2 echo "Setting CUDA versions for $VARIANT"
 if [[ $VARIANT == cu101* ]]; then
     CUDA_VERSION='10.1.105-1'
-    CUDA_PATCH_VERSION='10.1.105-1'
+    CUDA_PATCH_VERSION='10.1.0.105-1'
```
Comment (stu1130) [inline tools/setup_gpu_build_tools.sh]: @szha I found a typo here please review it. thanks

Comment (stu1130) [reply tools/setup_gpu_build_tools.sh]: I will rerun the bechmark again tonight and update the data on #14887

**Comment by stu1130** [comment]
ping @szha @perdasilva @marcoabreu for review

**Comment by stu1130** [comment]
@szha @marcoabreu could you merge this if it looks good to you? Thanks

**Comment by piyushghai** [comment]
@marcoabreu Gentle ping...

**Comment by Roshrini** [comment]
@marcoabreu @szha Could you take a look at this PR again? Thanks

**Comment by karan6181** [comment]
@marcoabreu @szha Gentle ping!

**Comment by marcoabreu** [comment]
Curious if this is gonna miss anything, considering the age of the PR. We'll see...

**Pull request merged**